Feat/updates

tescalada opened the pull request:
Bumping the version up to the latest version (9.1)

Adding the ability to set the image name via .env file so that the code doesn't need to be overwritten in each fork.

Fixing/adding the encr.sh script to the entrypoint so it can be used from the outside to encrypt passwords and set them in the kettle.properties file.

tescalada committed:
Fixing some typos in the docs/comments

just cleaning up some typos

tescalada committed:
Bumping PDI version to 9.1

bumping up to the latest version.

tescalada committed:
Switching keyservers

The sks keyserver is deprecated.
Switched to keyserver.ubuntu.com and that seems to have fixed the issue.

tescalada committed:
Add .env file for image name

This will enable setting a .env file with IMAGE=username/pdi in it to
override the image name in both the spoon script and the makefile.
When building or running in the makefile it will use the custom image
name and when using the spoon script it will use the custom image name.
This way any forks from here out can set a .env file with their own repo
name instead of needing to hardcode their own username into the code.
It should also allow for CI scripts to build and publish containers a
bit more easily.

This patch does not change the default behavior of the makefile or spoon
script, however the makefile's default image name is pdi, and the spoon
scripts default image name is tgmti/pdi. This should maybe be
standardized to the same default value at some point.

tescalada committed:
removing reference to docker-compose

docker-compose is no longer a requirement of the project

tescalada committed:
adding encr command to entrypoint

This will allow you to encrypt a password using the encr.sh script

dnrce commented on spoon:
```diff
@@ -1,7 +1,15 @@
+# pull in any overrides to IMAGE from the .env file
+if [ -f .env ]
+then
+  export $(cat .env | sed 's/#.*//g' | xargs)
```
"useless use of cat" -- `sed` can be passed `.env` as a target filename directly

I also don't think `xargs` is doing anything here. `$()` already flattens newlines into spaces.

tescalada replied:
fixed

dnrce commented on README.md:
```diff
@@ -5,7 +5,7 @@ Docker Pentaho Data Integration
 DockerFile for [Pentaho Data Integration](https://sourceforge.net/projects/pentaho/) (a.k.a kettel / PDI)
 
-This image is intendend to allow execution os PDI transformations and jobs throught command line and run PDI's UI (`Spoon`). PDI server (`Carter`) is available on this image.
+This image is intended to allow execution of PDI transformations and jobs through command line and run PDI's UI (`Spoon`). PDI server (`Carter`) is available on this image.
```
While you're making corrections to this line, how about updating `Carter` to the correct name of `Carte`?

tescalada replied:
done

dnrce commented on Dockerfile:
```diff
@@ -35,6 +35,9 @@ RUN apt-get update && apt-get install -y xauth libwebkitgtk-1.0-0
 	&& kitchen.sh -file samples/transformations/files/test-job.kjb
 
+# Copy plugins to the plugins directory
+COPY ./plugins/ plugins/
```
Rather than supporting forks as variants, would it be a more standard workflow for variants to build a dependent image that uses this image in `FROM`? Additional plugins would be added as part of the dependent Dockerfile.

tescalada replied:
That was my original plan. I started down that path with another image building off of this image to include plugins but then the makefile and spoon file which now use the .env image override are in the original repo, not in whatever repo you built your new image in. So at that point you need to go back to the original repo to copy the spoon file at least, maybe the makefile. Overall it just started to feel messy as I went down that road and I decided to go back to just having a branch with my plugins bundled in so that the .env file, makefile, and spoon file all worked as expected. Does that make sense, or do you think the plugins should still be in a dependent image, maybe just within this repo? Both images could be built and tagged appropriately and pushed to dockerhub.

tescalada replied:
I ended up figuring out how to make this work with dependent images, which makes more sense. They are now just going to be in this repo and defined in the .env file. Still built using the make command. Seems like a better solution. Still not perfect, but better, probably.

dnrce commented on README.md:
```diff
@@ -19,6 +19,7 @@ Usage:	/entrypoint.sh COMMAND
 Pentaho Data Integration (PDI)
 
 Options:
+  encr password		Encrypt a password
```
This should say "obfuscate" rather than "encrypt" (here and in the help text).

tescalada replied:
done

tescalada committed:
Add the ability to build a custom dockerfile

This allows building of custom dockerfiles that extend the main image
without needing to fork the repo. The DOCKERFILE name can be set in the
.env file so that you can just run make setup to build your custom
image. The reason to build a custom image is to include custom plugins
in the build.
